src/Microsoft.ML.Data/Data/SchemaDefinition.cs

```diff
@@ -405,40 +407,44 @@ public static SchemaDefinition Create(Type userType, Direction direction = Direc
+                            else
+                                itemType = new KeyDataViewType(dataType, keyAttr.KeyCount.Count.GetValueOrDefault());
+                        }
                         else
-                            itemType = new KeyDataViewType(dataType, keyAttr.KeyCount.Count.GetValueOrDefault());
-                    }
-                    else
-                        itemType = ColumnTypeExtensions.PrimitiveTypeFromType(dataType);
-
-                    var vectorAttr = memberInfo.GetCustomAttribute<VectorTypeAttribute>();
-                    if (vectorAttr != null && !isVector)
-                        throw Contracts.ExceptParam(nameof(userType), $"Member {memberInfo.Name} marked with {nameof(VectorTypeAttribute)}, but does not appear to be a vector type", memberInfo.Name);
-                    if (isVector)
-                    {
-                        int[] dims = vectorAttr?.Dims;
-                        if (dims != null && dims.Any(d => d < 0))
-                            throw Contracts.ExceptParam(nameof(userType), "Some of member {0}'s dimension lengths are negative");
-                        if (Utils.Size(dims) == 0)
-                            columnType = new VectorDataViewType(itemType, 0);
+                            itemType = ColumnTypeExtensions.PrimitiveTypeFromType(dataType);
+
+                        var vectorAttr = memberInfo.GetCustomAttribute<VectorTypeAttribute>();
+                        if (vectorAttr != null && !isVector)
+                            throw Contracts.ExceptParam(nameof(userType), $"Member {memberInfo.Name} marked with {nameof(VectorTypeAttribute)}, but does not appear to be a vector type", memberInfo.Name);
+                        if (isVector)
+                        {
+                            int[] dims = vectorAttr?.Dims;
+                            if (dims != null && dims.Any(d => d < 0))
+                                throw Contracts.ExceptParam(nameof(userType), "Some of member {0}'s dimension lengths are negative");
+                            if (Utils.Size(dims) == 0)
+                                columnType = new VectorDataViewType(itemType, 0);
+                            else
+                                columnType = new VectorDataViewType(itemType, dims);
+                        }
                         else
-                            columnType = new VectorDataViewType(itemType, dims);
+                            columnType = itemType;
                     }
                     else
-                        columnType = itemType;
+                        columnType = DataViewTypeManager.GetDataViewType(dataType, customAttributes);
                 }
-                else
-                    columnType = DataViewTypeManager.GetDataViewType(dataType, customAttributes);
+                else columnType = CiphertextDataViewType.Instance;
```

Comment:
You shouldn't be adding this hard-coded type here. Instead, look at how we use the `DataViewTypeManager` above to allow for extension types.